src/main/java/frc/robot/OI.java

```diff
@@ -0,0 +1,235 @@
+package frc.robot;
+
+import edu.wpi.first.wpilibj.Joystick;
+import edu.wpi.first.wpilibj.XboxController;
+import edu.wpi.first.wpilibj.GenericHID.RumbleType;
+import edu.wpi.first.wpilibj.shuffleboard.BuiltInLayouts;
+import edu.wpi.first.wpilibj.shuffleboard.Shuffleboard;
+import edu.wpi.first.wpilibj.shuffleboard.ShuffleboardLayout;
+import edu.wpi.first.wpilibj2.command.button.JoystickButton;
+import java.util.Map;
+import java.util.function.Supplier;
+
+public class OI {
+  // Operator Interface (OI) class containing all control information
+
+  private static final int driverJoystickPort = 0;
+  private static final int operatorJoystickPort = 1;
+
+  public static final class Driver {
+
+    private static enum Button {
```

Comment:
is most of this class needed given the existence of the CommandXboxController class?

Reply:
This class was created to allow controls to be displayed on a shuffleboard tab - so the driver is never confused when programming changes them

Reply:
I could rewrite as a wrapper on the xboxcontroller class? This is pretty much exactly cut from last year

Reply:
if there is value-added compared to the CommandXboxController, and the code is already written I dont have any complaints. I just want to make sure we are using this code for a reason